### PR TITLE
[Tracking] Composite increase performance

### LINKED ISSUE #1728: [Bug][perf] CompositeTracker collects product data even if no actual tracker is enabled
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no

I did something like this:

```php
    public function isEnabled()
    {
        if ($this->enabled !== null) {
            return $this->enabled;
        }
        
        foreach ($this->trackerRegistry as $tracker) {
            if ($tracker->isEnabled()) {
                $this->enabled = true;
                
                return true;
            }
        }
        
        return $this->enabled = false;
    }
    ```
    
Then each call to `track*()` in the composite tracker checks 

```php
        if (!$this->isEnabled()) {
            return;
        }
```

skipping an expensive `extractTrackingData` which is not used.

### PR DESCRIPTION
**Pull request by dpfaffenbauer**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  |no
| BC breaks?    | no
| Deprecations? |no
| Fixed tickets | fixes #1728

@dkarlovi like this?

### CODE REVIEW

**Review by dkarlovi — APPROVED** [review]
LGTM.

File: src/CoreShop/Component/Tracking/Tracker/CompositeTracker.php [hunk 1]
```diff
@@ -19,13 +19,27 @@
...
-        return true;
+        if ($this->enabled !== null) {
+            return $this->enabled;
+        }
+
+        foreach ($this->trackerRegistry as $tracker) {
+            if ($tracker->isEnabled()) {
+                $this->enabled = true;
+
+                return true;
```
Comment (jdreesen) [inline src/CoreShop/Component/Tracking/Tracker/CompositeTracker.php:38]:
```suggestion
                return $this->enabled = true;
```

File: src/CoreShop/Component/Tracking/Tracker/CompositeTracker.php [hunk 2]
```diff
@@ -31,7 +31,7 @@ public function isEnabled(): bool
             return $this->enabled;
         }
 
-        foreach ($this->trackerRegistry as $tracker) {
+        foreach ($this->trackerRegistry->all() as $tracker) {
```
Comment (dkarlovi) [inline src/CoreShop/Component/Tracking/Tracker/CompositeTracker.php:34]: SA, the most boring colleague you want to have. :)

**Review by dkarlovi — APPROVED** [review]

File: src/CoreShop/Component/Tracking/Tracker/CompositeTracker.php (hunk 2 above)
Comment (dpfaffenbauer) [reply src/CoreShop/Component/Tracking/Tracker/CompositeTracker.php:34]: ;)

**Pull request merged**